wrappers/ds_selinux_restorecon.sh.in

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+# BEGIN COPYRIGHT BLOCK
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# All rights reserved.
+#
+# License: GPL (version 3 or any later version).
+# See LICENSE for details.
+# END COPYRIGHT BLOCK
+
+# Make sure we have the path to the dse.ldif
+if [ -z $1 ]
+then
+    echo "usage: ${0} /etc/dirsrv/slapd-<instance>/dse.ldif"
+    exit 1
+fi
+
+# Grep the db_home_dir out of the config file
+DS_HOME_DIR=`grep 'nsslapd-db-home-directory: ' $1 | awk '{print $2}'`
+
+# Now run restorecon
+restorecon ${DS_HOME_DIR:?}
```

Comment:
Note that `restorecon` (as part of `policycoreutils`) is not installed by default on OpenSUSE (only with `--recommends`). On Debian/Ubuntu 389-ds-base packages pull this dependency, but not on OpenSUSE and perhaps on some other distros that don't use SELinux. I'm not sure if there is a better way to fix this other than notify maintainers to patch another file.

Reply:
I just pushed a change to at least check if restorecon is available before calling it.  Another option is using a python script (python-policycoreutils & python-libselinux).  We require it in our specfile, but I guess that doesn't help people who use their own custom specfile.

Reply:
Thanks, with this change it doesn't prevent instance to start if `restorecon` is not available.